drivers/adc/adc_ifx_cat1.c

```diff
@@ -0,0 +1,271 @@
+		return -EINVAL;
+	}
+
+	if (sequence->oversampling) {
+		LOG_ERR("Oversampling not supported");
+		return -ENOTSUP;
+	}
+
+	int return_val = validate_buffer_size(sequence);
+
+	if (return_val < 0) {
+		LOG_ERR("Invalid sequence buffer size");
+		return return_val;
+	}
+
+	data->channels = channels;
+	data->buffer = sequence->buffer;
+	adc_context_start_read(&data->ctx, sequence);
+
+	return adc_context_wait_for_completion(&data->ctx);
+}
+
+static int ifx_cat1_adc_read(const struct device *dev,
+			     const struct adc_sequence *sequence)
+{
+	int ret;
+	struct ifx_cat1_adc_data *data = dev->data;
+
+	adc_context_lock(&data->ctx, false, NULL);
+	ret = start_read(dev, sequence);
+	adc_context_release(&data->ctx, ret);
+	return ret;
+}
+
+#ifdef CONFIG_ADC_ASYNC
+static int ifx_cat1_adc_read_async(const struct device *dev,
+				   const struct adc_sequence *sequence,
+				   struct k_poll_signal *async)
+{
+	int ret;
+	struct ifx_cat1_adc_data *data = dev->data;
+
+	adc_context_lock(&data->ctx, true, async);
+	ret = start_read(dev, sequence);
+	adc_context_release(&data->ctx, ret);
+
+	return ret;
+}
+#endif
+
+static int ifx_cat1_adc_init(const struct device *dev)
+{
+	struct ifx_cat1_adc_data *data = dev->data;
+	const struct ifx_cat1_adc_config *config = dev->config;
+	cy_rslt_t result;
+
+	data->dev = dev;
+
+	/* Initialize ADC. The ADC block which can connect to the input pin is selected */
+	result = cyhal_adc_init(&data->adc_obj, CYHAL_GET_GPIO(_ADCSAR_PORT, 0), NULL);
+	if (result != CY_RSLT_SUCCESS) {
+		LOG_ERR("ADC initialization failed. Error: 0x%08X\n", (unsigned int)result);
+		return -EIO;
+	}
+
+	/* Enable ADC Interrupt */
+	cyhal_adc_enable_event(&data->adc_obj, (cyhal_adc_event_t)ADC_CAT1_EVENTS_MASK,
+			       config->irq_priority, true);
+	cyhal_adc_register_callback(&data->adc_obj, _cyhal_adc_event_callback, (void *) dev);
+
+	adc_context_unlock_unconditionally(&data->ctx);
+
+	return 0;
+}
+
+static const struct adc_driver_api adc_cat1_driver_api = {
+	.channel_setup = ifx_cat1_adc_channel_setup,
+	.read = ifx_cat1_adc_read,
+	#ifdef CONFIG_ADC_ASYNC
+	.read_async = ifx_cat1_adc_read_async,
+	#endif
+	.ref_internal = ADC_CAT1_REF_INTERNAL_MV
+};
+
+/* Macros for ADC instance declaration */
+#define INFINEON_CAT1_ADC_INIT(n)					       \
+	static struct ifx_cat1_adc_data ifx_cat1_adc_data##n = {	       \
+		ADC_CONTEXT_INIT_TIMER(ifx_cat1_adc_data##n, ctx),	       \
+		ADC_CONTEXT_INIT_LOCK(ifx_cat1_adc_data##n, ctx),	       \
+		ADC_CONTEXT_INIT_SYNC(ifx_cat1_adc_data##n, ctx),	       \
+	};								       \
+									       \
+	static const struct ifx_cat1_adc_config adc_cat1_cfg_##n = {	       \
+		.irq_priority = DT_INST_IRQ(n, priority),		       \
+	};								       \
+									       \
+	DEVICE_DT_INST_DEFINE(n, ifx_cat1_adc_init,			       \
+			      NULL, &ifx_cat1_adc_data##n,		       \
+			      &adc_cat1_cfg_##n,			       \
+			      POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEVICE, \
```

Comment:
```suggestion
			      POST_KERNEL, CONFIG_ADC_INIT_PRIORITY,	       \
```
See b0cdef3c25c2ac92627f158700efdb9203ecb51a.

Reply:
Done.  Thanks for pointing this out.